File: groupy_api/Api_groupy/WebApplication1/Interfaces/ICidadeRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using WebApplication1.Models;
using WebApplication1.Models.Localidade;

namespace WebApplication1.Interfaces
{
    public interface ICidadeRepository
    {
        Task<List<Cidade>> GetAll();
        Task<Cidade> GetById(int cidadeId);
        Task<List<Cidade>> GetByUf(string uf);
        Task<Cidade> GetByUser(int userId);

    }
}

File: groupy_appV3/groupy_app/lib/repositories/localidade/cidade/icidade_repository.dart
import 'package:groupy_app/models/cidade_model.dart';

abstract class ICidadeRepository {
  Future<List<Cidade>> getAll();
  Future<Cidade?> getOne(String id);
  Future<Cidade?> getByUser(String userId);
  Future<List<Cidade>> getByUf(String uf);
}
